src/nnet3/nnet-test-utils.cc

```diff
@@ -804,6 +804,24 @@ static void GenerateRandomComponentConfig(std::string *component_type,
       os << "input-dim=" << input_dim << " output-dim=" << output_dim;
       break;
     }
+    case 17: {
+      *component_type = "Convolutional1dComponent";
+      int32 patch_stride = 10 + Rand() % 50, patch_step = 1 + Rand() % 4, 
+	    patch_dim = 4 + Rand () % 5;
+
+      // decrease patch_stride until (patch_stride - patch_dim) % patch_step == 0
+      while ((patch_stride - patch_dim) % patch_step != 0)
```

Comment:
Done. One issue: I left the previous 2 commits (by david and vijay) in the interactive editor when using rebase, so currently there are 3 commits in this pull request. But actually I didn't modify anything in their commits. I am not sure if it is fine for you to merge only mine. If not, I can create a new pull request. Sorry about it.
Yiming